### PR TITLE
Add apical point function

### PR DESCRIPTION
**Pull request by lidakanari**
This algorithm, proposed by @arnaudon solves the problem of not having apical point distance, as it always returns a result. This is useful for synthesis purposes.

### CODE REVIEW

File: tmd/Topology/analysis.py [hunk 1]
```diff
@@ -190,6 +190,25 @@ def histogram_stepped(ph1):
...
+        min_bin = np.min(ph_2D)
+    if max_bin is None:
+        max_bin = np.max(ph_2D)
+    bins = np.linspace(min_bin, max_bin, num_bins, dtype=float)
+    bin_centers = 0.5 * (bins[:-1] + bins[1:])
+    data = []
+    for i in range(num_bins -1):
+        for p in ph1:
+            if np.max(p) > bins[i+1] and np.min(p) < bins[i]:
+                data.append(bin_centers[i])
```
Comment (arnaudon) [inline tmd/Topology/analysis.py]: Would that work? 

`[bin_centers[i] for p in ph1 for i in range(num_bins - 1) if np.max(p) > bins[i+1] and np.min(p) < bins[i]]`

Comment (lidakanari) [reply tmd/Topology/analysis.py]: Yep, almost (ph1 -> ph_2D) ;)

Comment (arnaudon) [reply tmd/Topology/analysis.py]: Ah this is your change of notation haha

Comment (mgeplf) [reply tmd/Topology/analysis.py]: why not use `np.histogram`?

Comment (arnaudon) [reply tmd/Topology/analysis.py]: Because the function kde = stats.gaussian_kde(data) wants data not histrogram values

Comment (mgeplf) [reply tmd/Topology/analysis.py]: Ah, sorry, I meant `np.digitize`

Comment (arnaudon) [reply tmd/Topology/analysis.py]: I didn't know that one, it seems to be doing the job indeed! Thanks!

**Pull request merged**